editor/plugins/texture_region_editor_plugin.cpp

```diff
@@ -480,20 +480,41 @@ void TextureRegionEditor::_region_input(const Ref<InputEvent> &p_input) {
 			Vector2 dragged(mm->get_relative().x / draw_zoom, mm->get_relative().y / draw_zoom);
 			hscroll->set_value(hscroll->get_value() - dragged.x);
 			vscroll->set_value(vscroll->get_value() - dragged.y);
-
 		} else if (drag) {
 			if (edited_margin >= 0) {
 				float new_margin = 0;
-				if (edited_margin == 0) {
-					new_margin = prev_margin + (mm->get_position().y - drag_from.y) / draw_zoom;
-				} else if (edited_margin == 1) {
-					new_margin = prev_margin - (mm->get_position().y - drag_from.y) / draw_zoom;
-				} else if (edited_margin == 2) {
-					new_margin = prev_margin + (mm->get_position().x - drag_from.x) / draw_zoom;
-				} else if (edited_margin == 3) {
-					new_margin = prev_margin - (mm->get_position().x - drag_from.x) / draw_zoom;
+
+				if (snap_mode != SNAP_GRID) {
+					if (edited_margin == 0) {
```

Comment:
It's a bit weird that those checks are done based on ints and not enums, but that's for another PR :)

Reply:
Just following how it's done everywhere else in the file. :stuck_out_tongue: